### DISCUSSION TITLE
Can't compile simple server example

### DISCUSSION THREAD

**Discussion by pimin9** [mathieucarbou/AsyncTCP#35] [Q&A]
Arduino-ide (2.3.4) compiling for  LOLIN D32 PRO I get the following errors;

C:\Users\Paul Newhouse.DESKTOP-MBT5GN1\AppData\Local\Temp\.arduinoIDE-unsaved20241129-17316-154c159.k0mk\SimpleServer\SimpleServer.ino:113:1: error: 'LoggingMiddleware' does not name a type
  113 | LoggingMiddleware requestLogger;
      | ^~~~~~~~~~~~~~~~~
C:\Users\Paul Newhouse.DESKTOP-MBT5GN1\AppData\Local\Temp\.arduinoIDE-unsaved20241129-17316-154c159.k0mk\SimpleServer\SimpleServer.ino:116:1: error: 'CorsMiddleware' does not name a type
  116 | CorsMiddleware cors;
      | ^~~~~~~~~~~~~~
... many more

Where did I go wrong?

Thanks,
Paul

**Comment by mathieucarbou** [comment]
I don't know... Too few details to assert anything ;-)

**Reply by pimin9** [reply]
Ok.  I'm trying to compile a supplied example.  What details do you need?

Am I supposed to write LoggingMiddleWare?

**Reply by mathieucarbou** [reply]
It's like you miss a library but I am sorry I won't be able to help you : I am using a real IDE (vscode + pioarduino) which is capable of correctly handling transitive dependencies and projet descriptors.

Arduino IDE ina toy which let's the user handle all these things manually, so this is sadly more error prone and there are countless of ways to mess up using Arduino IDE

Both AsyncTCP and ESPAsyncWS have samples that are tested in CI so I know they build fine on Arduino core 2.x and 3.x and on many boards ;-)

**Comment by pimin9** [comment]
Agreed Adruino-ide is  .. ??ummmm???  let's say primitive ?  Will look into
other alternatives.  I was moving along pretty well until I started using
it for Web things.

On Sun, Dec 29, 2024 at 4:05 PM Mathieu Carbou ***@***.***>
wrote:

> It's like you miss a library but I am sorry I won't be able to help you :
> I am using a real IDE (vscode + pioarduino) which is capable of correctly
> handling transitive dependencies and projet descriptors.
>
> Arduino IDE ina toy which let's the user handle all these things manually,
> so this is sadly more error prone and there are countless of ways to mess
> up using Arduino IDE
>
> Both AsyncTCP and ESPAsyncWS have samples that are tested in CI so I know
> they build fine on Arduino core 2.x and 3.x and on many boards ;-)
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/mathieucarbou/AsyncTCP/discussions/35#discussioncomment-11693000>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AHWJ6WIZRQLUWFHYJNXQGZD2ICE5BAVCNFSM6AAAAABULESKWOVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCNRZGMYDAMA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
I pledge allegiance to liberty and justice for all.
Liberty and Justice for Ukraini
